src/main/java/graphql/schema/GraphQLFieldDefinition.java

```diff
@@ -20,27 +20,28 @@
     private final String name;
     private final String description;
     private GraphQLOutputType type;
-    private final DataFetcher dataFetcher;
+    private final DataFetcherFactory dataFetcherFactory;
     private final String deprecationReason;
     private final List<GraphQLArgument> arguments;
     private final FieldDefinition definition;
 
 
+    @Deprecated
     @Internal
-    public GraphQLFieldDefinition(String name, String description, GraphQLOutputType type, DataFetcher dataFetcher, List<GraphQLArgument> arguments, String deprecationReason) {
-        this(name,description,type, dataFetcher,arguments,deprecationReason,null);
+    public GraphQLFieldDefinition(String name, String description, GraphQLOutputType type, DataFetcher<?> dataFetcher, List<GraphQLArgument> arguments, String deprecationReason) {
+        this(name, description, type, DataFetcherFactory.useDataFetcher(dataFetcher), arguments, deprecationReason, null);
     }
 
     @Internal
-    public GraphQLFieldDefinition(String name, String description, GraphQLOutputType type, DataFetcher dataFetcher, List<GraphQLArgument> arguments, String deprecationReason, FieldDefinition definition) {
+    public GraphQLFieldDefinition(String name, String description, GraphQLOutputType type, DataFetcherFactory dataFetcherFactory, List<GraphQLArgument> arguments, String deprecationReason, FieldDefinition definition) {
         assertValidName(name);
-        assertNotNull(dataFetcher, "dataFetcher can't be null");
+        assertNotNull(dataFetcherFactory, "dataFetcherFactory can't be null");
```

Comment:
this is confusing I think: until know we only have `DataFetcher` ... no we say `DataFetcherFactory` can't be null: maybe something like 'you have to provide a DataFetcher (or DataFetcherFactory)' ?

Reply:
done